assets/src/edit-story/components/library/panes/media/local/mediaPane.js

```diff
@@ -139,24 +142,54 @@ function MediaPane(props) {
         setMediaType,
         setSearchTerm,
         uploadVideoPoster,
+        optimizeVideo,
       };
     }
   );
 
   const { showSnackbar } = useSnackbar();
+  const enableMediaPickerVideoOptimization = useFeature(
+    'enableMediaPickerVideoOptimization'
+  );
 
   const {
+    allowedAllowedTranscodableTypes,
     allowedFileTypes,
     allowedMimeTypes: {
       image: allowedImageMimeTypes,
       video: allowedVideoMimeTypes,
     },
   } = useConfig();
 
-  const allowedMimeTypes = useMemo(
-    () => [...allowedImageMimeTypes, ...allowedVideoMimeTypes],
-    [allowedImageMimeTypes, allowedVideoMimeTypes]
-  );
+  const { isFeatureEnabled, isTranscodingEnabled } = useFFmpeg();
+
+  const allowedMimeTypes = useMemo(() => {
+    if (
+      enableMediaPickerVideoOptimization &&
+      isFeatureEnabled &&
+      isTranscodingEnabled
+    ) {
+      return [
+        ...allowedAllowedTranscodableTypes,
+        ...allowedImageMimeTypes,
+        ...allowedVideoMimeTypes,
+      ];
+    }
+    return [...allowedImageMimeTypes, ...allowedVideoMimeTypes];
+  }, [
+    allowedImageMimeTypes,
+    allowedVideoMimeTypes,
+    isFeatureEnabled,
+    isTranscodingEnabled,
+    enableMediaPickerVideoOptimization,
+    allowedAllowedTranscodableTypes,
+  ]);
+
+  const transcodableMimeTypes = useMemo(() => {
+    return allowedAllowedTranscodableTypes.filter(
+      (x) => !allowedVideoMimeTypes.includes(x)
+    );
+  }, [allowedAllowedTranscodableTypes, allowedVideoMimeTypes]);
```

Comment:
Remove video/mp4 and video/webm from list of transcodable mime types here. We don't need to do anything special for these file types, they should just work.